pkg/liqo-controller-manager/webhooks/foreigncluster/fc.go

```diff
@@ -49,30 +63,63 @@ func (w *fcwh) DecodeForeignCluster(obj runtime.RawExtension) (*discoveryv1alpha
 	return &fc, err
 }
 
-// Handle implements the ForeignCluster validating webhook logic.
+// Handle implements the ForeignCluster validating webhook logic for create operations.
 //
 //nolint:gocritic // The signature of this method is imposed by controller runtime.
-func (w *fcwh) Handle(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation != admissionv1.Update {
-		return admission.Allowed("")
-	}
-
-	// In case of updates, prevent the mutation of the PeeringType field.
-	fcnew, err := w.DecodeForeignCluster(req.Object)
+func (w *fcwhm) Handle(ctx context.Context, req admission.Request) admission.Response {
+	fc, err := w.DecodeForeignCluster(req.Object)
 	if err != nil {
 		klog.Errorf("Failed decoding ForeignCluster object: %v", err)
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	fcold, err := w.DecodeForeignCluster(req.OldObject)
-	if err != nil {
-		klog.Errorf("Failed decoding ForeignCluster object: %v", err)
-		return admission.Errored(http.StatusBadRequest, err)
+	if fc.Spec.PeeringType == "" {
+		return admission.Denied("The PeeringType value cannot be empty")
 	}
 
-	if fcold.Spec.PeeringType != fcnew.Spec.PeeringType {
-		return admission.Denied("The PeeringType value cannot be modified after creation")
+	// set cluster-id label to easy retrieve ForeignClusters by ClusterId,
+	// if it is added manually, the name maybe not coincide with ClusterId
+	if fc.ObjectMeta.Labels == nil {
+		fc.ObjectMeta.Labels = map[string]string{}
 	}
+	fc.ObjectMeta.Labels[discovery.ClusterIDLabel] = fc.Spec.ClusterIdentity.ClusterID
 
+	marshaledFc, err := json.Marshal(fc)
+	if err != nil {
+		klog.Errorf("Failed marshaling ForeignCluster object: %v", err)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledFc)
+}
+
+// Handle implements the ForeignCluster validating webhook logic for update operations.
+//
+//nolint:gocritic // The signature of this method is imposed by controller runtime.
+func (w *fcwhv) Handle(ctx context.Context, req admission.Request) admission.Response {
+	if req.Operation == admissionv1.Update {
```

Comment:
It is more idiomatic to adopt a return early policy, to prevent indenting the core logic. Additionally, it is unclear to me why you modified the logic that already existed.